keeper/query_server_portal.go

```diff
@@ -67,6 +70,36 @@ func (k portalQueryServer) Peers(ctx context.Context, req *portal.QueryPeers) (*
 	return &portal.QueryPeersResponse{Peers: peers}, err
 }
 
+func (k portalQueryServer) SupportedBridgingPaths(ctx context.Context, req *portal.QuerySupportedBridgingPaths) (*portal.QuerySupportedBridgingPathsResponse, error) {
+	if req == nil {
+		return nil, types.ErrInvalidRequest
+	}
+
+	// NOTE: We have to typecast here because protoc-gen-grpc-gateway doesn't
+	// support it via gogoproto.
+	destinationChainId := uint16(req.DestinationChainId)
+
+	iterator, err := k.PortalSupportedBridgingPaths.Iterate(
```

Comment:
does the iterator need to be closed?

Reply:
Yes it does, good catch! But actually, @g-luca recommended switching to the `Walk` functionality.

So I have made that change in https://github.com/noble-assets/dollar/pull/21/commits/6e29d58f27ad92809033bc54e57d1c5e5e6766d0.